youtube_dl/extractor/pornhub.py

```diff
@@ -22,10 +22,14 @@
     remove_quotes,
     str_to_int,
     url_or_none,
+    urlencode_postdata,
 )
 
 
 class PornHubBaseIE(InfoExtractor):
+
+    _NETRC_MACHINE = 'pornhub'  # or 'pornhubpremium'
```

Comment:
Does premium login also work on pornhub.com site? In any case it's better to have separate extractors with separate `_NETRC_MACHINE` with common base class in case one may want to use different credentials.

Reply:
No. The account logins are completely separate. You can have a pornhub.com and a pornhubpremium.com account that are totally separate.

I took the approach of having separate extractors before but that lead to more duplicated code. The extractors are exactly the same except for the login process.

Reply:
If you look at [line 54](https://github.com/ytdl-org/youtube-dl/pull/24294/files#diff-79cdff5086134653cc40744315017e07R54) you'll see we load the appropriate credentials based on the URL.

Reply:
Then extractors must be separate. Move common code in the base class and there will be no duplicated code.

Reply:
> Move common code in the base class and there will be no duplicated code. &mdash; @dstftw 

Yeah, the only way to do this would be to move all the code from the subclasses (playlist extractors) up into the base class and then creating a new `PornHubPremiumIE` that extends from `PornHubBaseIE`.

That will create a much larger pull-request. Are you comfortable reviewing that? I avoided that approach because I didn't think a larger pull would be accepted.

This right here is the simplest and most straightforward change and IMO the best way to implement this.

Reply:
An earlier attempt I made tried this approach. Please take a quick look here and see if you're okay with this direction:

https://github.com/twaddington/youtube-dl/blob/phpremium/youtube_dl/extractor/pornhubpremium.py

Note that there is still some duplicated code.

Reply:
The class structure would look something like this:

```
PornHubBaseIE
  - PornHubIE
    - PornHubUserIE
    - PornHubPagedVideoListIE
    - PornHubUserVideosUploadIE
  - PornHubPremiumIE
    - PornHubPremiumUserIE
    - PornHubPremiumPagedVideoListIE
    - PornHubPremiumUserVideosUploadIE
```